### PR TITLE
update runbot to support lucene recognizer

### PR DESCRIPTION
**Pull request by coldplaying42**
Fixes #

## Proposed Changes

  - 
  -
  -

### CODE REVIEW

File: experimental/generation/runbot/Startup.cs [hunk 1]
```diff
@@ -51,6 +54,16 @@ public void ConfigureServices(IServiceCollection services)
...
+            ComponentRegistration.Add(new DeclarativeComponentRegistrationBridge<DynamicListBotComponent>());
+            ComponentRegistration.Add(new DeclarativeComponentRegistrationBridge<LuisBotComponent>());
+/*
+            new DialogsBotComponent().ConfigureServices(services, this._configuration);
+            new DeclarativeBotComponent().ConfigureServices(services, this._configuration);
+            new AdaptiveBotComponent().ConfigureServices(services, this._configuration);
+            new LanguageGenerationBotComponent().ConfigureServices(services, this._configuration);
+            new QnAMakerBotComponent().ConfigureServices(services, this._configuration);
+            new LuisBotComponent().ConfigureServices(services, this._configuration);
+            new DynamicListBotComponent().ConfigureServices(services, this._configuration);*/
```
Comment (coldplaying42) [inline experimental/generation/runbot/Startup.cs:67]: When I use the new-version registration codes, the bot could not be correctly started in the emulator. Do I miss something for this._configuration?

File: experimental/generation/generator/packages/library/templates/standard/form-BeginDialog.dialog.lg [hunk 2]
```diff
@@ -30,9 +30,9 @@
 ```
 
 # setPropertiesForDynamiclists
-- IF: ${!empty(where(schema.properties, property, property.value.$template == "dynamiclist"))}
+- IF: ${!empty(where(schema.properties, property, property.value.$template == "dynamiclist" || property.value.$template == "LuceneDynamicList" ))}
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/standard/form-BeginDialog.dialog.lg:33]: > || property.value.$template == "LuceneDynamicList"  [](http://example.com/codeflow?start=91&length=52)

You can't make this change--it presumes knowledge that these templates don't have.  You could revamp this to use a transformer to add the set properties and use the transformer as part of the lucene templates.

**Review by chrimc62 — CHANGES_REQUESTED** [review]
:clock1:

File: experimental/generation/generator/packages/library/templates/standard/form-BeginDialog.dialog.lg (hunk 2 above)
Comment (chrimc62) [reply experimental/generation/generator/packages/library/templates/standard/form-BeginDialog.dialog.lg:33]: You still can't make this change--revert it.

File: experimental/generation/runbot/Startup.cs [hunk 3]
```diff
@@ -45,12 +47,22 @@ public void ConfigureServices(IServiceCollection services)
...
-            ComponentRegistration.Add(new LanguageGenerationComponentRegistration());
-            ComponentRegistration.Add(new QnAMakerComponentRegistration());
-            ComponentRegistration.Add(new LuisComponentRegistration());
+            ComponentRegistration.Add(new DeclarativeComponentRegistrationBridge<DialogsBotComponent>());
+            ComponentRegistration.Add(new DeclarativeComponentRegistrationBridge<DeclarativeBotComponent>());
+            ComponentRegistration.Add(new DeclarativeComponentRegistrationBridge<AdaptiveBotComponent>());
+            ComponentRegistration.Add(new DeclarativeComponentRegistrationBridge<LanguageGenerationBotComponent>());
+            ComponentRegistration.Add(new DeclarativeComponentRegistrationBridge<QnAMakerBotComponent>());
+            ComponentRegistration.Add(new DeclarativeComponentRegistrationBridge<LuisBotComponent>());
+            ComponentRegistration.Add(new DeclarativeComponentRegistrationBridge<DynamicListBotComponent>());
```
Comment (chrimc62) [inline experimental/generation/runbot/Startup.cs:56]: This works fine, but you only really needed to do this for your dynamic list bot component.

File: experimental/generation/runbot/RunBot.csproj [hunk 4]
```diff
@@ -28,14 +28,15 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Iciclecreek.Bot.Builder.Dialogs.Recognizers.LuceneZY" Version="4.13.0" />
```
Comment (chrimc62) [inline experimental/generation/runbot/RunBot.csproj:31]: >LuceneZY [](http://example.com/codeflow?start=75&length=8)

You can't publish the standard package?

Comment (coldplaying42) [reply experimental/generation/runbot/RunBot.csproj:31]: When the form-operations.en-us.data issue is resolved, I will publish the standard package. And e2e test it.

Comment (coldplaying42) [reply experimental/generation/runbot/RunBot.csproj]: I think for nuget.org, different onwers (tom and zoma) could not have the nuget pkg with the same name. So I rename the nuget pkg to Iciclecreek.Bot.Builder.Dialogs.Recognizers.Lucene.Dynamiclist.

**Review by chrimc62 — APPROVED** [review]
:shipit:

File: experimental/generation/runbot/RunBot.csproj [hunk 5]
```diff
@@ -28,14 +28,15 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Iciclecreek.Bot.Builder.Dialogs.Recognizers.Lucene.Dynamiclist" Version="4.13.0" />
```
Comment (chrimc62) [inline experimental/generation/runbot/RunBot.csproj:31]: >Dialogs [](http://example.com/codeflow?start=55&length=7)

I would drop this in the namespace.

**Comment by chrimc62** [comment]
I'm surprised there is not a schema update--I know the schema should have changed.

**Pull request merged**